lib/find_non_repeated_letter.rb

```diff
@@ -9,6 +9,28 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
+  
+  array = str.chars
+  
+  final = array.length - 1
+  
+  answer = ""
+  
+  array.each_index do |i|
+    if i == 0 && array[i] != array[(i+1)]
+      answer = array[i]
+      break
+    elsif i == final && array[i] != array[(i-1)]
+      answer = array[i]
+      break
+    elsif array[i] != array[(i-1)] && array[i] != array[(i+1)]
```

Comment:
If I run my code as-is (with the loop) and take out that line, the rake fails.

I wrote out the conditionals that needed to be met in English and then kind of translated it to code. 
We should return the object and break the loop if:
We are in the first index of the array and the letter is not the same at the next
We are at the final index of the array and the letter is not the same as the previous
Or we are anywhere else in the array and the letter is not the same as to either the previous or next letter.

I *think* as written, removing `array[i] != array[(i-1)]` means that something like "aabbddc" would return "a".